### PR TITLE
initial specs for project req

### PR DESCRIPTION
**Pull request by tildeee**
Things to do:

- wordsmith
- more optionals
- feedback.md
- PR template

i'm putting my solution on a branch called ddr/soln, and is in this PR: #2

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,186 @@
...
+
+You should write your program as a series of statements within a file `generator.rb`. Anyone with this file should be able to run the command `$ ruby generator.rb` to run the program.
+
+## Implementation Requirements
+
+When a user runs this program, the program should politely ask the user for the following things:
+
+1. a word to combine to begin the portmanteau
+1. a second word to combine and begin the portmanteau
+
```
Comment (droberts-sea) [inline README.md:42]: Should the second word be "begin"?

File: README.md [hunk 2]
```diff
@@ -1 +1,186 @@
...
+#### `run_generator` method
+
+Make a method named `run_generator`. This method has no parameters. Start this method by outputting a prompt to ask for a first input word A, and accept some input. Then, ask for a second input word B, and accept some input.
+
+Make sure you save these inputs into variables. Then, output these variables by printing them to the console, and the statement `"in the run_generator method"`.
+
+Then, _call_ or _invoke_ the `run_generator` outside of the method definition, at the bottom of the `generator.rb` file.
+
+Confirm that this is all setup by running `$ ruby generator.rb`, and make sure you can input two words, and you eventually see the statement `"in the run_generator method"` in the console.
```
Comment (droberts-sea) [inline README.md:66]: Maybe provide a sample execution?

File: README.md [hunk 3]
```diff
@@ -1 +1,186 @@
...
+
+In wave two, work on combining the first two input words like portmanteau rules.
+
+Implement these rules:
+
+#### Input validation rules
+
+- The input for the first word must be at least two characters. If the input is less than 2 characters (as in 0 or 1 characters), the program should handle it in the following method:
+  - output a message to the command line that informs the user that the input was invalid and must be at least 2 characters
+  - asks the user to re-enter the input for the same prompt
```
Comment (droberts-sea) [inline README.md:114]: Line 99: "method" is a keyword, did you intend to use it that way?

Comment (droberts-sea) [reply README.md:114]: Is this more difficult than the error checking we had them do in Calculator?

File: README.md [hunk 4]
```diff
@@ -1 +1,186 @@
...
+There is no requirement that inputs must be upper or lowercase.
+
+#### Combination rules
+
+- Vowels in this program are the following letters: `a`, `e`, `i`, `o`, or `u`, and nothing else
+- In the first word, find the last instance of a vowel. The first half of the portmanteau will keep every letter of the first word until and **excluding** its last vowel.
+
+  For example, if the first word is `dreams`, then `a` is the last vowel in this word. The first part of the new portmanteau is `dre`, excluding the `a`
+- In the second word, find the first instance of a vowel. The second half of the portmanteau will keep every letter of the second word until and **including** its last vowel.
+
```
Comment (droberts-sea) [inline README.md:132]: Should line 116 read "after and **including** its first vowel"?

**Review by droberts-sea — CHANGES_REQUESTED** [review]
Looks good in general. I am a little concerned about the difficulty level - this feels harder than calculator overall, and is a 1-day project very early on. It's also much more scaffolded, so maybe it's fine. Worth thinking about though.

File: README.md (hunk 3 above)
Comment (tildeee) [reply README.md:114]: @droberts-ada wrt the question about error checking: Ideally, this means:

```ruby
input = gets.chomp
until ( input.length > 1 )
  puts "the input was invalid and must be at least 2 characters"
  input = gets.chomp
end
```

What do you think?

**Pull request merged**